src/gflip/gflip_engine.hpp

```diff
@@ -320,6 +362,27 @@ class gflip_engine
                        std::cout<<"Dictionary dimension "<<dictionary_dimensions<<std::endl;
 		}
 		
+		/*
+		 * Serialize the TF_IDF and scan_bows to an archive
+		 * @param o_archive output archive to serialize to
+		*/
+		void save(boost::archive::binary_oarchive& o_archive) const
+		{
+		               o_archive << laserscan_bow;
+		               o_archive << tf_idf;
+		}
+
+		/*
+		 * Loads TF_IDF and scan_bows from a boost archive
+		 * @param i_archive input archive to load from
+		*/
+		void load(boost::archive::binary_iarchive& i_archive)
+		{
+		               laserscan_bow.clear();
```

Comment:
Sometimes the boost serialization appends at the end of the stl container instead of replacing depending on whether the type "is_default_constructible". So it's better to clear.